# Simple Text Analysis Using TextBlob

We will use TextBlob to perform simple sentiment analysis on an online article. <br>Any article can be used for this, but I'm using a Wikipedia page about [Python.](https://en.wikipedia.org/wiki/Python_(programming_language)

## Step 1: Install + Import Necessary Libraries

In [ ]:
# !pip install textblob > This is the library we will use to perform basic NLP tasks
# !pip install newspaper3k > This is the library we will use to scrape + extract information from online resources

In [1]:
from textblob import TextBlob
from newspaper import Article 

## Step 2: Reading Data

In [2]:
# Any url/article can be used for this analysis

url = 'https://en.wikipedia.org/wiki/Python_(programming_language)'

## Step 3: Exploring + Preparing Data

In [3]:
article = Article(url)

article.download()
article.parse()
article.nlp()

In [5]:
text = article.summary # You can use entire text, but we will only use summary section for this analysis
text

'General-purpose programming languagePython is a high-level, general-purpose programming language.\n[35][36]Guido van Rossum began working on Python in the late 1980s as a successor to the ABC programming language and first released it in 1991 as Python 0.9.0.\nReleases of Python 3 include the 2to3 utility, which automates the translation of Python 2 code to Python 3.\nObject-oriented programming and structured programming are fully supported, and many of their features support functional programming and aspect-oriented programming (including metaprogramming[68] and metaobjects).\nPython 3.10 deprecated wstr (to be removed in Python 3.12; meaning Python extensions[172] need to be modified by then),[173] and added pattern matching to the language.'

## Step 4: Sentiment Analysis

In [6]:
blob = TextBlob(text)

sentiment = blob.sentiment.polarity

The above code will give us a score that falls between **-1 to +1.** <br>
-1 means the algorithm has detected a "negative" tone in the text, while +1 means a "positive" tone. The closer the score is to zero, the more "neutral" the algorithm deems the analyzed text to be.

In [7]:
print(sentiment)

0.15


## Step 5: Results

The result for the above article is 0.15, which makes sense given that it's just a Wikipedia page about Python. <br>
It's important to note that this is a very basic method of performing sentiment analysis. The results of this model aren't very accurate overall, and they don't perform well when it comes to detecting sarcasm, humour etc., <br>
For more advanced analysis, I recommend using a more complex model like Roberta's Transformers.